source/extensions/filters/http/ext_proc/processor_state.cc

```diff
@@ -139,137 +139,167 @@ ProcessorState::getCallbackStateAfterHeaderResp(const CommonResponse& common_res
-        // the original one.
-        headers_->removeContentLength();
-        body_replaced_ = true;
-        if (bufferedData() == nullptr) {
-          Buffer::OwnedImpl new_body;
-          MutationUtils::applyBodyMutations(common_response.body_mutation(), new_body);
-          addBufferedData(new_body);
-        } else {
-          modifyBufferedData([&common_response](Buffer::Instance& buf) {
-            MutationUtils::applyBodyMutations(common_response.body_mutation(), buf);
-          });
-        }
-      }
+  if (common_response.status() == CommonResponse::CONTINUE_AND_REPLACE) {
+    return handleHeadersResponseWithContinueAndReplace(common_response);
+  }
+
+  return handleHeadersResponseWithContinue();
+}
 
-      // In case any data left over in the chunk queue, clear them.
-      clearStreamingChunk();
-      // Once this message is received, we won't send anything more on this request
-      // or response to the processor. Clear flags to make sure.
-      body_mode_ = ProcessingMode::NONE;
-      send_trailers_ = false;
+absl::Status
+ProcessorState::handleHeadersResponseWithContinueAndReplace(const CommonResponse& common_response) {
+  ENVOY_LOG(debug, "Replacing complete message");
+
+  if (common_response.has_body_mutation()) {
+    applyReplacementBodyMutation(common_response);
+  }
+
+  // Clear any leftover data and disable further processing
+  clearStreamingChunk();
+  body_mode_ = ProcessingMode::NONE;
+  send_trailers_ = false;
+  clearWatermark();
+
+  return finalizeResponse();
+}
+
+void ProcessorState::applyReplacementBodyMutation(const CommonResponse& common_response) {
+  // Remove content length since it won't match the new body
+  headers_->removeContentLength();
+  body_replaced_ = true;
+
+  if (bufferedData() == nullptr) {
+    Buffer::OwnedImpl new_body;
+    MutationUtils::applyBodyMutations(common_response.body_mutation(), new_body);
+    addBufferedData(new_body);
+  } else {
+    modifyBufferedData([&common_response](Buffer::Instance& buf) {
+      MutationUtils::applyBodyMutations(common_response.body_mutation(), buf);
+    });
+  }
+}
+
+absl::Status ProcessorState::handleHeadersResponseWithContinue() {
+  if (no_body_) {
+    ENVOY_LOG(debug, "The message had no body");
+    return finalizeContinueResponse();
+  }
+
+  if (complete_body_available_ && body_mode_ != ProcessingMode::NONE) {
+    return handleCompleteBodyAvailable();
+  }
+
+  switch (body_mode_) {
+  case ProcessingMode::BUFFERED:
+    return handleBufferedMode();
+  case ProcessingMode::STREAMED:
+    return handleStreamedMode();
+  case ProcessingMode::FULL_DUPLEX_STREAMED:
+    return handleFullDuplexStreamedMode();
+  case ProcessingMode::BUFFERED_PARTIAL:
+    return handleBufferedPartialMode();
+  default:
+    return finalizeContinueResponse();
+  }
+}
+
+absl::Status ProcessorState::handleCompleteBodyAvailable() {
+  if (callback_state_ == CallbackState::Idle) {
+    if (bufferedData()) {
```

Comment:
In the case of if (bufferedData()) returns false, the original code will call not return OkStatus() here. Instead, it will call finalizeContinueResponse().  This might be the reason of the CI failure.  PTAL!

Reply:
It's being returned [[Ref](https://github.com/envoyproxy/envoy/pull/37186/files#r1857322135)]. Due to refactoring it's a bit hard to see.